src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -86,6 +87,50 @@ public ProjectFactory(string msbuildDirectory, string path, IDictionary<string,
             Initialize(project);
         }
 
+        // This handler is called only when the common language runtime tries to bind to the assembly and fails
+        private Assembly AssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            if (string.IsNullOrEmpty(_msbuildDirectory))
+            {
+                return null;
+            }
+
+            var failingAssemblyFilename = args.Name.Substring(0, args.Name.IndexOf(","));
```

Comment:
Is the comma guaranteed?

It would be safer to do ``args.Name.Split(",", RemoveEmpty).FirstOrDefault()`` here